js/core/renderer_base.js

```diff
@@ -156,6 +156,58 @@ initRender.prototype.toggleClass = function(className, value) {
     return this;
 };
 
+
+["width", "height", "outerWidth", "outerHeight", "innerWidth", "innerHeight"].forEach(function(funcName) {
+    var name = funcName.toLowerCase().indexOf("width") >= 0 ? "Width" : "Height";
+    var isOuter = funcName.indexOf("outer") === 0;
+    var isInner = funcName.indexOf("inner") === 0;
+
+    initRender.prototype[funcName] = function(value) {
+        var element = this[0];
+
+        if(!element) {
+            return;
+        }
+
+        if(typeUtils.isWindow(element)) {
+            return isOuter ? element["inner" + name] : element.document.documentElement["client" + name];
+        }
+
+        if(element.nodeType === Node.DOCUMENT_NODE) {
+            var documentElement = element.documentElement;
+
+            return Math.max(
+                element.body["scroll" + name],
+                element.body["offset" + name],
+                documentElement["scroll" + name],
+                documentElement["offset" + name],
+                documentElement["client" + name]
+            );
+        }
+
+        if(arguments.length === 0 || typeof value === "boolean") {
+            return sizeUtils["get" + name](element, { paddings: isInner || isOuter, borders: isOuter, margins: value });
+        }
+
+        if(value === undefined || value === null) {
+            return this;
+        }
+
+        var resultValue = typeUtils.isFunction(value) ? value() : value;
```

Comment:
We set size in overlay.js, the input value is option value of the component:

`this._$content.outerWidth(this.option("width"))`

Width and height values is setted as function by default:

`width: function() { return $(window).width() * 0.8; },`

I think it will be better to leave the condition here.